src/graph/impl/KokkosGraph_GraphColor_impl.hpp

```diff
@@ -2379,6 +2379,300 @@ class GraphColor_VB:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t
+    nnz_lno_persistent_work_view_t dependency
+      = nnz_lno_persistent_work_view_t("dependency", numVertices);
+    Kokkos::View<size_type, MyTempMemorySpace> frontierSize("newFrontier");
+    typename Kokkos::View<size_type, MyTempMemorySpace>::HostMirror host_frontierSize =
+      Kokkos::create_mirror_view(frontierSize);
+    Kokkos::View<size_type, MyTempMemorySpace> newFrontierSize("newFrontierSize");
+    typename Kokkos::View<size_type, MyTempMemorySpace>::HostMirror host_newFrontierSize =
+      Kokkos::create_mirror_view(newFrontierSize);
+    nnz_lno_temp_work_view_t frontier = nnz_lno_temp_work_view_t("frontier", numVertices);
+    nnz_lno_temp_work_view_t newFrontier = nnz_lno_temp_work_view_t("new frontier", numVertices);
+    Kokkos::parallel_for("Deterministic Coloring: compute dependency list", numVertices,
+                         KOKKOS_LAMBDA(const int node) {
+                           int myScore = score(node);
+                           int numNeighs = myXadj(node + 1) - myXadj(node);
+                           for(int neigh = 0; neigh < numNeighs; ++neigh) {
+                             if(myScore < score(myAdj(myXadj(node) + neigh))) {
+                               dependency(node) = dependency(node) + 1;
+                             }
+                             if(( myScore == score(myAdj(myXadj(node) + neigh)) ) &&
+                                ( node < myAdj(myXadj(node) + neigh) )) {
+                               dependency(node) = dependency(node) + 1;
+                             }
+                           }
+                           if(dependency(node) == 0) {
+                             const size_type newFrontierIdx
+                               = Kokkos::atomic_fetch_add(&newFrontierSize(), 1);
+                             newFrontier(newFrontierIdx) = node;
+                           }
+                         });
+
+    Kokkos::deep_copy(host_newFrontierSize, newFrontierSize);
+    while(host_newFrontierSize() > 0) {
+      ++num_loops;
+      // First swap fontier with newFrontier and fontierSize with newFrontierSize
+      // reset newFrontierSize
+      Kokkos::parallel_for("Swap frontier sizes", 1,
+			   KOKKOS_LAMBDA(const int dummy) {
+			     frontierSize() = newFrontierSize();
+			     newFrontierSize() = 0;
+			   });
+      Kokkos::deep_copy(host_frontierSize, frontierSize);
+      {
+	auto swap_tmp = frontier;
+	frontier = newFrontier;
+	newFrontier = swap_tmp;
+      }
+
+      // Loop over nodes in the frontier
+      // First variant without bit array, easier to understand/program
+      if(this->_use_color_set == 0) {
+        Kokkos::parallel_for("Deterministic Coloring: color nodes in frontier", host_frontierSize(),
+                             KOKKOS_LAMBDA(const size_type frontierIdx) {
+
+                               size_type frontierNode = frontier(frontierIdx);
+                               int* bannedColors = new int[maxColors];
```

Comment:
I still don't like this array. This is a crash waiting to happen and something that cannot perform well due to global memory access.  Can we check the node_type early in the code and issue a warning. Doesn't have to be in this PR, but better to do it before next master promotion.